core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -163,7 +161,7 @@ public void shutdownNow(Status status) {
     }
     if (savedPendingStreams != null) {
       for (PendingStream stream : savedPendingStreams) {
-        stream.setError(status);
+        stream.cancel(status);
```

Comment:
Status here would commonly be UNAVAILABLE, which is not permitted in ClientStream.cancel(). Let's just change ClientStream to allow anything other that `isOk()`?